### ISSUE TITLE
Add more inspectors

### ISSUE THREAD

**Issue by gakonst** [flashbots/mev-inspect-rs#10]
**Is your feature request related to a problem? Please describe.**

- https://github.com/flashbots/mev-inspect-ts/issues?q=is%3Aissue+is%3Aopen+%22inspector%22+

**Comment by obadiaa** [comment]
Single-tx inspectors to add:
- DyDx liquidations (@brockelmore)
- ESD and DSD clippers (@brockelmore)
- Set Protocol rebalances
- Nexus Mutual rebalances
- Balancer?
- Long tail arb?
- Inspector for bundle of transactions with same gas price?

Multi-tx inspectors to add:
- Uniswap sandwiching
- Backrunning with oracle updates (DyDx liquidations)
- Multi-legged arb?

**Comment by obadiaa** [comment]
Very naively as I'm still understanding how Inspect functions:

**Outstanding strategies to cover:**
Single-tx inspectors to add:
- DyDx liquidations (@brockelmore)
- ESD and DSD clippers (@brockelmore)
- Set Protocol rebalances
- Nexus Mutual rebalances
- Long tail arb?
- Inspector for bundle of transactions with same gas price?

**Multi-tx inspectors to add:**
- Uniswap sandwiching
- Maker liquidation
- Backrunning with oracle updates (DyDx liquidations)
- Multi-legged arb?

**Comment by gakonst** [comment]
Agreed with everything, clarifying:
* What does long tail arb mean?
* What is a multi-legged arb? Arb1 -> room for some other txs by other players -> Arb2?

**Comment by obadiaa** [comment]
Added some extra resources for the inspectors above and happy to brainstorm some more, although I feel @jparyani's work on taking the other approach of looking at what the top bots are doing and using that as a compass for which inspectors to write might be a lil better

@gakonst I added both of these points from the original ts repo issue you linked, I'm not sure what long-tail arb means in this context, maybe @thegostep remembers? 

Multi-legged arb just means when you go through several price pairs instead of just 2, but it can still be within the same transaction!

**Comment by thegostep** [comment]
I believe long tail means small arbs not profitable under normal execution risk conditions

**Comment by gakonst** [comment]
Yeah @obadiaa now that we have support for the "big" protocols, I agree using the big bots as a compass.

**Comment by obadiaa** [comment]
We could also use Gervais et. al's paper https://arxiv.org/pdf/2101.05511.pdf for some direction

Generally, protocols I can think of to cover:
- Swerve (?)
- Mooniswap
- Bancor
- Kyber

Seems like our inspectors already cover a large market share though

![image](https://user-images.githubusercontent.com/19510814/106155324-4e659680-615f-11eb-8757-740140151e75.png)
